src/System.Runtime/tests/System/CharTests.netcoreapp.cs

```diff
@@ -0,0 +1,168 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using System.Text.Unicode.Tests;
+using Xunit;
+using Xunit.Sdk;
+
+namespace System.Tests
+{
+    public partial class CharTests
+    {
+        [OuterLoop]
+        [Fact]
+        public static void GetUnicodeCategory_Char_AllInputs()
+        {
+            // This tests calls char.GetUnicodeCategory for every possible input, ensuring that
+            // the runtime agrees with the data in the core Unicode files.
+
+            for (uint i = 0; i <= char.MaxValue; i++)
```

Comment:
Why `uint` and not `char`? Is it because of the comment below?
(asking more out of interest)

Reply:
It'd be an infinite loop if `uint` were changed to `char` ;)

Reply:
Ups, embarrassing ;-) thanks!